plugins/woocommerce/includes/class-wc-post-data.php

```diff
@@ -405,13 +405,27 @@ public static function untrash_post( $id ) {
 			$data_store->untrash_variations( $id );
 
 			wc_product_force_unique_sku( $id );
+			self::clear_global_unique_id_if_necessary( $id );
 
 			wc_get_container()->get( ProductAttributesLookupDataStore::class )->on_product_changed( $id );
 		} elseif ( 'product_variation' === $post_type ) {
 			wc_get_container()->get( ProductAttributesLookupDataStore::class )->on_product_changed( $id );
 		}
 	}
 
+	/**
+	 * Clear global unique id if it's not unique.
+	 *
+	 * @param mixed $id Post ID.
+	 */
+	private static function clear_global_unique_id_if_necessary( $id ) {
+		$product = wc_get_product( $id );
+		if ( ! wc_product_has_global_unique_id( $id, $product->get_global_unique_id() ) ) {
```

Comment:
There should be a check that `$product` exists before attempting to call methods on it.